### PR TITLE
NOTIF-69 Separate RBAC for Notifications and Integrations.

### PR DESCRIPTION
**Pull request by pilhuhn**
This queries the RBAC server for both notifications and integrations data and then sets the roles accordingly.

The RBAC-server needs a configuration for integrations before this can be deployed.

### CODE REVIEW

**Review by josejulio — APPROVED** [review]

**Comment by josejulio** [comment]
We don't have integrations permissions on ci, but code looks good!

**Comment by theute** [comment]
There is a configurations for integrations but it's not platform defaukt and we are not account admin :-/
https://github.com/RedHatInsights/rbac-config/blob/master/configs/permissions/integrations.json
https://github.com/RedHatInsights/rbac-config/blob/master/configs/roles/integrations.json

**Comment by pilhuhn** [comment]
The demo is over, so we should probably tackle this.

File: src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java [hunk 1]
```diff
@@ -199,7 +200,7 @@
 
     @PUT
     @Path("/email/subscription/instant")
-    @RolesAllowed("write")
+    @RolesAllowed(RbacIdentityProvider.RBAC_WRITE_INTEGRATIONS)
```
Comment (pilhuhn) [inline src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java:203]: How should this be protected? The end-user would initiate this - no? But the end-user only has `integrations:endpoints:read`.
Same below for unsubscribe and the dailyEmail versions.
@theute @josejulio ?

Comment (theute) [reply src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java]: I guess email subscriptions should not be protected ? (beside only being able to edit his own settings)

Comment (josejulio) [reply src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java]: Notice we *maybe* don't need this endpoint, unless we want to provide additional flexibility. The user preferences pages uses the [UserConfigService](https://github.com/RedHatInsights/notifications-backend/blob/cdec897fadde7a8e45d130a346b278df2f5abb86/src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java#L28).

If we want to keep this, maybe only requiring the same permissions as the userpreference page does (only auth required?)

**Pull request merged**